File: ABC094/A - Cats and Dogs/catsanddogs.ts
import * as fs from 'fs';

// 入力を読み込み、トリム（前後の空白・改行削除）してから分割する
const input = fs.readFileSync(0, 'utf8').trim().split(/\s+/);

// 数値に変換（Number関数の方がparseIntより柔軟で一般的です）
const A = Number(input[0]);
const B = Number(input[1]);
const X = Number(input[2]);

if (A <= X && X <= A + B) {
    console.log("YES");
} else {
    console.log("NO");
}

File: ABC094/A - Cats and Dogs/catsanddogs.cs
using System;

// 1行の入力を読み込み、空白で分割して数値に変換する
string[] input = Console.ReadLine().Split(' ');
int A = int.Parse(input[0]);
int B = int.Parse(input[1]);
int X = int.Parse(input[2]);

// 条件分岐（論理演算子はC言語と同じ && を使用）
if (A <= X && X <= A + B)
{
    Console.WriteLine("YES");
}
else
{
    Console.WriteLine("NO");
}
